app.py

```diff
@@ -231,9 +233,26 @@ def before_request_https():
             # We use a 302 in case we need to revert the redirect.
             return redirect(url, code=302)
 
+def version():
+    """Get the version from the Heroku environment variables."""
+    if not os.getenv('DYNO'):
+        # Not on Heroku
+        return 'DEV'
+
+    vrz = os.getenv('HEROKU_RELEASE_CREATED_AT')
+    the_date = datetime.date.fromisoformat(vrz[:10]) if vrz else datetime.date.today()
+
+    commit = os.getenv('HEROKU_SLUG_COMMIT', '????')[0:6]
+    return the_date.strftime('%b %d') + f' ({commit})'
+
 # Unique random key for sessions.
 # For settings with multiple workers, an environment variable is required, otherwise cookies will be constantly removed and re-set by different workers.
-app.config['SECRET_KEY'] = os.getenv ('SECRET_KEY') or uuid.uuid4().hex
+if version() == 'DEV':
```

Comment:
Agree with everything you said. We should use `IS_PRODUCTION` to see if we're in `beta`, and distinguish between Heroku and non-Heroku.

I would suggest using a helper method (perhaps in `utils.py`) to determine if we're on Heroku, that returns the result of `__name__ == '__main__'`, for readability purposes. Maybe we can call it `is_heroku`?

In terms of which thing to enforce here, I'd make it so that if we're in Heroku, there should be a `SECRET KEY` already.

Reply:
@rix0rrr I went ahead and added `utils.is_heroku`, but we can of course revert to `__name__ == '__main__'`.